Graphics/Slicer/Math/Skeleton/Motorcycles.hs

```diff
@@ -142,9 +143,14 @@ concaveMotorcycles contour = catMaybes $ onlyMotorcycles <$> zip (linePairs cont
 -- | Find where a motorcycle intersects a contour, if the motorcycle is emitted from between the two given segments.
 --   If the motorcycle lands between two segments, return the second segment, as well.
 motorcycleIntersectsAt :: Contour -> Motorcycle -> (LineSeg, Maybe LineSeg)
-motorcycleIntersectsAt contour motorcycle@(Motorcycle (inSeg,outSeg) _)
-  | length (getMotorcycleIntersections motorcycle contour) == 2 && length foundSegEvents == 1 = head foundSegEvents
-  | otherwise = error $ "handle more than one intersection point here." <> show (getMotorcycleIntersections motorcycle contour) <> "\n"
+motorcycleIntersectsAt contour motorcycle@(Motorcycle (inSeg,outSeg) _) = case getMotorcycleIntersections motorcycle contour of
```

Comment:
Isn't replacing non-total functions with a case switch with lots of `error` cases cheating?  :)

What about making the type:

```haskell
motorcycleIntersectsAt :: Contour -> Motorcycle -> MonadError MotoIntersectError m => m (LineSeg, Maybe LineSeg)
```

with some

```haskell
data MotoIntersectError = Need2IntersectionsButGot Int | Need2SegEventsButGot Int
```

(I have to admit though that I'm not sure how much value that'll give you in hslice.  In latex it's fine to crash because all program runs terminate quickly one way or another; in a CRUD server it may be essential to handle errors in order to keep the state consistent.  hslice is probably more like latex.)

Or what about shaping the input in a way that makes the function (more) total, like [here](https://github.com/Haskell-Things/HSlice/blob/356070df7b7d0560977d1ac000300430853c5dd9/programs/extcuraengine.hs#L120-L127)?

Reply:
yeah, i'm fine with the errors and the dieing. this is cheating, but it's highly visible cheating.